ui-admin/src/study/surveys/SurveyEnvironmentTable.tsx

```diff
@@ -0,0 +1,129 @@
+import { StudyEnvParams } from '../StudyEnvironmentRouter'
+import {
+  ENVIRONMENT_NAMES,
+  EnvironmentName,
+  StudyEnvironmentSurvey,
+  StudyEnvironmentSurveyNamed
+} from '@juniper/ui-core'
+import React, { useState } from 'react'
+import { Link } from 'react-router-dom'
+import { Button, EllipsisDropdownButton } from '../../components/forms/Button'
+import SurveyEnvironmentDetailModal from './SurveyEnvironmentDetailModal'
+
+export type SurveyTableProps = {
+  stableIds: string[], // the stableIds of surveys to show in the table
+  studyEnvParams: StudyEnvParams,
+  configuredSurveys: StudyEnvironmentSurveyNamed[]
+  setSelectedSurveyConfig: (config: StudyEnvironmentSurvey) => void
+  showDeleteSurveyModal: boolean
+  setShowDeleteSurveyModal: (show: boolean) => void
+  showArchiveSurveyModal: boolean
+  setShowArchiveSurveyModal: (show: boolean) => void
+  updateConfiguredSurvey: (surveyConfig: StudyEnvironmentSurvey) => void
+}
+
+
+/**
+ * shows a list of surveys and the version currently live in each environment
+ */
+export default function SurveyEnvironmentTable(props: SurveyTableProps) {
+  if (props.stableIds.length === 0) {
+    return <div className="fst-italic fw-light pb-3 ps-2">None</div>
+  }
+  return <table className="table table-striped table-bordered">
```

Comment:
Is there a reason not to use tanstack react table here? You would have to generate the column definitions on the fly... but I think it could work.

Reply:
that's a good question.  I hadn't anticipated any need for sorting/filtering capability, but it can't hurt

Reply:
this was a great suggestion -- moving it to a tanstack table didn't add any functionality, but I think it made the code easier to follow